### PR TITLE
add initial pass of unit tests

### LINKED ISSUE #8: add unit tests for the project

### PR DESCRIPTION
**Pull request by edaniszewski**
This PR:
- adds a first-pass for unit tests, bringing coverage to ~75%

fixes #8

### CODE REVIEW

**Comment by edaniszewski** [comment]
bump for review

**Comment by marcoceppi** [comment]
Lint is currently failing, can you run `tox -e fmt` and commit the changes?

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    pytest<4.0.0
+    pytest-asyncio
+    pytest-cov>=2.5.1
+    pytest-mock>=1.6.0
+commands=
+    pytest -s \
+        --cov-report term \
+        --cov-report html \
+        --cov=prophetess \
+        {posargs}
```
Comment (marcoceppi) [inline tox.ini]: Can we add `-p no:warnings` on Python 3.8 asynctest still uses `@coroutine` wrappers which triggers a wall of DeprecationWarning

**Comment by edaniszewski** [comment]
👌 thanks for the review

I ran formatting and linting looks happy on my machine now. Also added back the `-p no:warnings` for pytest run. (:

**Pull request merged**